docs/demultiplexing.md

```diff
@@ -0,0 +1,171 @@
+# Demultiplexing in Directed Acyclic Graph Scheduler
+
+## Pipeline branching
+Directed Acyclic Graph Scheduler allows creating pipelines with optional parameter `demultiply_count: N` which adds ability to any node (`DL model` and `custom`) to slice outputs into `N` separate sub outputs and branch pipeline execution into `N` sub pipelines. Following nodes will be executed by event loop `N` times independently from each other and results will be gathered and  packed into one output just before sending a response. Additionally `gather_from_node: <node_name>` parameter can be used to manually specify to perform gathering at any point in Directed Acyclic Graph.
```

Comment:
Maybe if we use verb "branch" for demultiplication then use "converge" for gathering?

Reply:
`If no "gather_from_node" is specified then convering will be done automatically before sending the response.`
Maybe add

Reply:
"Results will be gathered" phrase is used here, "results will converge" sound odd. "Pipeline will converge" would  be better, but I'd like to stay with "results" here.

Reply:
`If no "gather_from_node" is specified then convering will be done automatically before sending the response.` - this is already implicated by phrase `Additionally `gather_from_node: <node_name>` parameter can be used to...`

Reply:
Additionaly, few lines below there is something similar:
```
This example has no `gather_from_node` parameter specified in config.json configuration file. This means, gathering (shown in diagram below) will be done automatically by `response` node at the end of event loop processing.
````